Common/include/toolboxes/graph_toolbox.hpp

```diff
@@ -592,3 +628,23 @@ struct DummyGridColor
   inline IteratorLikeInt begin() const {return IteratorLikeInt(0);}
   inline IteratorLikeInt end() const {return IteratorLikeInt(size);}
 };
+
+
+/*!
+ * \brief Computes the efficiency of a grid coloring for given number of threads and chunk size.
+ */
+template<class SparsePattern>
+su2double coloringEfficiency(const SparsePattern& coloring, int numThreads, int chunkSize)
+{
+  using Index_t = typename SparsePattern::IndexType;
+
+  /*--- Ideally compute time is proportional to total work over number of threads. ---*/
+  su2double ideal = coloring.getNumNonZeros() / su2double(numThreads);
+
+  /*--- In practice the total work is quantized first by colors and then by chunks. ---*/
+  Index_t real = 0;
+  for(Index_t color = 0; color < coloring.getOuterSize(); ++color)
+    real += chunkSize * roundUpDiv(roundUpDiv(coloring.getNumNonZeros(color), chunkSize), numThreads);
```

Comment:
The computation of coloring efficiency is described here, it is just a simple heuristic.